llvm/lib/CodeGen/SelectionDAG/LegalizeVectorOps.cpp

```diff
@@ -1070,13 +1071,21 @@ void VectorLegalizer::Expand(SDNode *Node, SmallVectorImpl<SDValue> &Results) {
     break;
   case ISD::FMINNUM:
   case ISD::FMAXNUM:
+    if (SDValue Expanded = ExpandBF16Arith(Node)) {
```

Comment:
Probably shouldn't just bypass the dedicated function to lower something

Reply:
I was a little unsure how generic `expandFMINNUM_FMAXNUM` actually is.  The description seemed pretty specific alongside a use in SIISelLowering, so wasn't sure if I could just add the splitting code into it.

Reply:
I've moved the expansion function into TargetLowering and updated the two expandFMIN###_FMAXNUM### functions to make us of it.